### PR TITLE
Remove updater and replace logic in update value propagation

### PR DESCRIPTION
**Pull request by gordoncaleb**
This PR removes the value updater on the vertex and puts the logic in the value propagation algorithms as it is a concept that only they are concerned about. This also puts a `T calculate()` method on all non-probabilistic vertices that can be expanded in the future to abstract out the graph state (i.e `T calculate(NetworkState state)`).

### CODE REVIEW

File: keanu-project/src/main/java/io/improbable/keanu/vertices/generic/nonprobabilistic/CPTVertex.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public OUT sample(KeanuRandom random) {
         return vertex == null ? defaultResult.sample(random) : vertex.sample(random);
     }
 
-    public OUT getDerivedValue() {
+    public OUT calculate() {
```
Comment (migwellian) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/generic/nonprobabilistic/CPTVertex.java]: please add `@Override`

File: keanu-project/src/main/java/io/improbable/keanu/vertices/bool/nonprobabilistic/BooleanIfVertex.java [hunk 2]
```diff
@@ -16,7 +16,7 @@ public BooleanIfVertex(int[] shape,
                            Vertex<? extends BooleanTensor> predicate,
                            Vertex<? extends BooleanTensor> thn,
                            Vertex<? extends BooleanTensor> els) {
-        super(new NonProbabilisticValueUpdater<>(v -> ((BooleanIfVertex) v).op(predicate.getValue(), thn.getValue(), els.getValue())));
+        super();
```
[the remark this answers is not in the document]
Comment (christophernorth) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/bool/nonprobabilistic/BooleanIfVertex.java]: 👍

File: keanu-project/src/main/java/io/improbable/keanu/vertices/Observation.java [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
 
+import java.util.Optional;
+
 public class Observation<T> implements Observable<T> {
     // package private - because it's created by the factory method Observable.observableTypeFor
-    Observation() {}
+    Observation() {
+    }
 
     private T observedValue = null;
```
[the remark this answers is not in the document]
Comment (gordoncaleb) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/Observation.java]: Good catch.

File: keanu-project/src/main/java/io/improbable/keanu/vertices/generic/nonprobabilistic/CPTVertex.java (hunk 1 above)
Comment (gordoncaleb) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/generic/nonprobabilistic/CPTVertex.java]: Nice catch!

File: keanu-project/src/main/java/io/improbable/keanu/vertices/Observation.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (migwellian) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/Observation.java]: Fair enough :-)

**Comment by migwellian** [comment]
PR looks good to me now :-)

**Pull request merged**